src/storage/src/render/debezium.rs

```diff
@@ -159,13 +157,50 @@ where
         } => panic!("render_tx should be called with a dedup mode"),
     };
 
-    // TODO(#11666): Should we try to avoid moving everything to one worker?
+    let DebeziumTransactionMetadata {
+        tx_data_collections_idx,
+        tx_data_collections_data_collection_idx,
+        tx_data_collection_name,
+        tx_data_collections_event_count_idx,
+        tx_status_idx,
+        tx_transaction_id_idx,
+        data_transaction_id_idx,
+        tx_metadata_global_id: _,
+    } = tx_metadata_description;
+
+    let tx_dist = move |&(ref row, time, _diff): &(Row, Timestamp, Diff)| match row
+        .iter()
+        .nth(tx_transaction_id_idx)
+    {
+        Some(Datum::String(s)) => s.hashed(),
+        // If there's no transaction_id then we won't be able to use this row so aim to distribute evenly
+        _ => time.hashed(),
+    };
+
+    let data_dist = move |result: &DecodeResult| {
+        // If we can't pull out the transaction_id, it doesn't matter which worker we end up on.  Position seems a good
+        // way to spread around the work as it comes in and is likely uncorrelated to the hash.
```

Comment:
we are already distributing the decode for debezium by `partition`, might want to stick with that? https://github.com/MaterializeInc/materialize/blob/main/src/storage/src/decode/mod.rs#L369-L372

Reply:
@cjubb39 hmm thats a good point, my understanding was the dedup logic depended on the partition local-ness of data, but I could be remembering wrong, @petrosagg may know...

Reply:
The requirement for debezium dedup logic is that the same logical message ends up on the same worker. This means that it is safe to distribute the records on any field that remains the same in the case of repetitions. Practically this means that from `SourceOutput` we can use the `key`, `value`, `partition` fields (or subsets of those) just fine. But `position` and `upstream_time_millis` are not the same when a message gets repeated so we can't use those for debezium processing. 

After writing the above I realize that we can change the distribution function to be based on `value` during decoding and remove the [special case for debezium](https://github.com/MaterializeInc/materialize/blob/main/src/storage/src/decode/mod.rs#L369-L372).

And in the debezium tx operator we should distribute based on transaction id and fallback on the value if that doesn't exist to have good distribution